pkg/cmd/server/start/start_master.go

```diff
@@ -265,11 +265,24 @@ func (o MasterOptions) RunMaster() error {
 		return kerrors.NewInvalid("MasterConfig", o.ConfigFile, validationResults.Errors)
 	}
 
-	if err := StartMaster(masterConfig); err != nil {
-		return err
+	if !o.MasterArgs.StartControllers {
+		masterConfig.Controllers = configapi.ControllersDisabled
 	}
 
-	return nil
+	if !o.MasterArgs.StartAPI {
+		openshiftConfig, err := origin.BuildMasterConfig(*masterConfig)
+		if err != nil {
+			return err
+		}
+
+		_, kubeConfig, err := buildKubernetesMasterConfig(openshiftConfig)
+		if err != nil {
+			return err
+		}
+		return StartControllers(openshiftConfig, kubeConfig)
```

Comment:
Addressed naming and defaulting behavior for StartAPI

On Thu, Aug 13, 2015 at 2:55 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/start/start_master.go
> <https://github.com/openshift/origin/pull/4132#discussion_r37011600>:
>
> >  	}
> >
> > -	return nil
> > +	if !o.MasterArgs.StartAPI {
> > +		openshiftConfig, err := origin.BuildMasterConfig(*masterConfig)
> > +		if err != nil {
> > +			return err
> > +		}
> > +
> > +		_, kubeConfig, err := buildKubernetesMasterConfig(openshiftConfig)
> > +		if err != nil {
> > +			return err
> > +		}
> > +		return StartControllers(openshiftConfig, kubeConfig)
>
> If we keep the separate StartControllers, I think I'm inclined to also
> make a separate StartAPIServer and call them directly from the commands.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4132/files#r37011600>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift